File: Python/Level-1(lists)/binary_search.py
a = []
n = int(input())
key = int(input())
for _ in range(n):
    a.append(int(input()))

def binary_search(a,n,key):
    low =0
    mid =0
    high = n-1
    while low <= high:
        mid = low + (high-low)//2
        if a[mid] == key:
            return 0
        elif a[mid] < key:
            low =mid +1
        else:
            high =mid -1
    return -1

val = binary_search(a,n,key)
if val == 0:
    print("Element is found")
else:
    print("Element is not found")

File: C/Easy(arrays)/binary_search.c
#include<stdio.h>

int binary_search(int a[],int n, int key)
{
    int low = 0;
    int mid;
    int high = n-1;
    while(low<=high)
    {
        mid = low + (high-low)/2;
        if(a[mid]==key)
        {
            return 0;
        }
        else if (a[mid] < key)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
        
    }
    return -1;
}

int main()
{
    int a[100],n,key;
    scanf("%d%d",&n,&key);
    for(int i = 0; i<n;i++)
    {
        scanf("%d",&a[i]);
    }
    int val = binary_search(a,n,key);
    if(val == 0)
    {
        printf("Element is found");
    }
    else
    {
        printf("Element is not found");
    }

    return 0;
}